### PR TITLE
Consistency of ContentDecodePolicy

### PR DESCRIPTION
**Pull request by lmazuel**
To bring closure on https://github.com/Azure/azure-sdk-for-python/issues/6331

Make the ContentDecodePolicy closer to its initial design (and then closer to the docstring that were still there :)

@johanste There is indeed a breaking change in some class method of ContentDecodePolicy, but I would be surprised it could have a major impact being how special this usage is.

### CODE REVIEW

File: sdk/core/azure-core/azure/core/pipeline/policies/universal.py [hunk 1]
```diff
@@ -341,32 +350,39 @@ def _json_attemp(data):
...
+
+        Use str and headers to NOT use any requests/aiohttp or whatever
         specific implementation.
         Headers will tested for "content-type"
 
-        :param response: The HTTP response.
-        :type response: ~azure.core.pipeline.transport.HttpResponse
+        :param content: Content to decode, could be text or bytes/stream (will be decoded with UTF8)
+        :type content: str or bytes or stream
+        :param str content_type: Raw content type from headers. Will be check for charset.
```
Comment (bryevdv) [inline sdk/core/azure-core/azure/core/pipeline/policies/universal.py]: check -> checked

File: sdk/core/azure-core/azure/core/pipeline/policies/universal.py [hunk 2]
```diff
@@ -278,17 +280,24 @@ class ContentDecodePolicy(SansIOHTTPPolicy):
     CONTEXT_NAME = "deserialized_data"
 
     @classmethod
-    def deserialize_from_text(cls, response, content_type=None):
-        # type: (Type[ContentDecodePolicyType], HTTPResponseType, Optional[str]) -> Any
+    def deserialize_from_text(
+        cls,  # type: Type[ContentDecodePolicyType]
+        data,  # type: Optional[Union[AnyStr, IO]]
+        content_type=None,  # Optional[str]
+        response=None  # Optional[_HttpResponseBase]
```
Comment (bryevdv) [inline sdk/core/azure-core/azure/core/pipeline/policies/universal.py]: FWIW it's a little odd that a documented type in a public API is a private class

Comment (lmazuel) [reply sdk/core/azure-core/azure/core/pipeline/policies/universal.py]: I could do `Union[HttpResponse, AsyncHttpResponse]` instead. More verbose, but at least not a private type, I agree

Comment (lmazuel) [reply sdk/core/azure-core/azure/core/pipeline/policies/universal.py]: fixed

File: sdk/core/azure-core/azure/core/pipeline/policies/universal.py (hunk 1 above)
Comment (lmazuel) [reply sdk/core/azure-core/azure/core/pipeline/policies/universal.py]: fixed

**Pull request merged**